Libraries/LibJS/Runtime/ObjectConstructor.cpp

```diff
@@ -165,4 +168,40 @@ Value ObjectConstructor::is(Interpreter& interpreter)
     return typed_eq(interpreter, value1, value2);
 }
 
+Value ObjectConstructor::keys(Interpreter& interpreter)
+{
+    if (!interpreter.argument_count())
+        interpreter.throw_exception<TypeError>("can't convert undefined to object");
+
+    auto* obj_arg = interpreter.argument(0).to_object(interpreter.heap());
+    if (interpreter.exception())
+        return {};
+
+    return obj_arg->get_enumerable_own_properties(*obj_arg, GetOwnPropertyMode::Key);
+}
+
+Value ObjectConstructor::values(Interpreter& interpreter)
+{
+    if (!interpreter.argument_count())
+        interpreter.throw_exception<TypeError>("can't convert undefined to object");
+
+    auto* obj_arg = interpreter.argument(0).to_object(interpreter.heap());
+    if (interpreter.exception())
+        return {};
+
+    return obj_arg->get_enumerable_own_properties(*obj_arg, GetOwnPropertyMode::Value);
+}
+
+Value ObjectConstructor::entries(Interpreter& interpreter)
+{
+    if (!interpreter.argument_count())
+        interpreter.throw_exception<TypeError>("can't convert undefined to object");
```

Comment:
We should return immediately here instead of continuing.
```suggestion
        return interpreter.throw_exception<TypeError>("Can't convert undefined to object");
```